optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -931,12 +931,9 @@ public int getEntityCount(Solution_ solution) {
      * @param scoreDirector never null
      * @return {@code >= 0}
      */
-    public int getMovableEntityCount(ScoreDirector<Solution_> scoreDirector) {
+    public boolean hasMovableEntities(ScoreDirector<Solution_> scoreDirector) {
         return extractAllEntitiesStream(scoreDirector.getWorkingSolution())
-                .mapToInt(entity -> findEntityDescriptorOrFail(entity.getClass()).isMovable(scoreDirector, entity)
-                        ? 1
-                        : 0)
-                .sum();
+                .anyMatch(entity -> findEntityDescriptorOrFail(entity.getClass()).isMovable(scoreDirector, entity));
```

Comment:
This is a beneficial change on its own. But looking at `findEntityDescriptorOrFail(...)`, it has a performance issue of its own.

It does try to cache the results of the lookup. Unfortunately, when nothing is found, it doesn't store anything in the cache - so in that case (= no entity descriptor found) the lookups will be happening over and over again.

If we're touching this code, I'd vote to address this issue too.

Reply:
@triceo Done. Please take a look.